File: web/tests/reference/dynamic-programming/partition-equal-subset-sum.js
function canPartition(nums) {
  const total = nums.reduce((a, b) => a + b, 0);
  if (total % 2 !== 0) return false;
  const target = total / 2;
  const reachable = new Array(target + 1).fill(false);
  reachable[0] = true;
  for (const x of nums) {
    for (let t = target; t >= x; t--) {
      if (reachable[t - x]) reachable[t] = true;
    }
  }
  return reachable[target];
}

File: web/tests/reference/dynamic-programming/partition-equal-subset-sum.py
def can_partition(nums):
    total = sum(nums)
    if total % 2 != 0:
        return False
    target = total // 2
    reachable = [False] * (target + 1)
    reachable[0] = True
    for x in nums:
        for t in range(target, x - 1, -1):
            if reachable[t - x]:
                reachable[t] = True
    return reachable[target]
